pkg/mimir/signals.go

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package mimir
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/grafana/dskit/services"
+	"github.com/weaveworks/common/signals"
+	"go.uber.org/atomic"
+)
+
+// newShutdownSignalReceiver creates a new signals.SignalReceiver implementation that
+// handles everything required to cleanly shut down Mimir in response to a SIGTERM or
+// SIGINT signal.
+func newShutdownSignalReceiver(delay time.Duration, ready *atomic.Bool, server *http.Server, manager *services.Manager) signals.SignalReceiver {
```

Comment:
For the same reason...

```suggestion
func newShutdownSignalReceiver(delay time.Duration, shutdownRequested *atomic.Bool, server *http.Server, manager *services.Manager) signals.SignalReceiver {
```